crates/common/mqtt_channel/src/config.rs

```diff
@@ -140,6 +155,16 @@ impl Config {
 
         mqtt_options.set_max_packet_size(self.max_packet_size, self.max_packet_size);
 
+        if let Some(lwp) = self.last_will_message.clone() {
```

Comment:
```suggestion
        if let Some(lwp) = self.last_will_message {
```

Reply:
Since the `last_will_message` is behind a shared reference, we have to clone it here. Another way is to take the ref `&self.last_will_message` but the problem is when the struct members (topic, message) are initialized, they have to be cloned because they are behind the shared ref.